### PR TITLE
Adding TS code foundations

### PR DESCRIPTION
**Pull request by rachfop**
## What does this PR do?
Completes Foundations sections for TS.

### CODE REVIEW

File: docs-src/typescript/how-to-customize-activity-type-in-typescript.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+sidebar_label: Customize Activity Type
+description: Customize Activity Type
+tags:
+  - developer-guide
+  - typescript
+---
+
+In TypeScript, to customize an Activity Name set the function name to your desired name.
+
+In the following example the Activity Name is `greet`.
```
Comment (lorensr) [inline docs-src/typescript/how-to-customize-activity-type-in-typescript.md:13]:
```suggestion
In TypeScript, the Activity Type is the function name.

In the following example, the Activity Type is `greet`.
```

you can't customize this in TS

**Review by lorensr — APPROVED** [review]

File: docs-src/typescript/how-to-customize-activity-type-in-typescript.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+<!--SNIPSTART typescript-activity-fn -->
+<!--SNIPEND-->
```
Comment (lorensr) [inline docs-src/typescript/how-to-customize-activity-type-in-typescript.md:16]:
```suggestion
In TypeScript, the Activity Type is the property of the [`WorkerOptions.activities`](https://typescript.temporal.io/api/interfaces/worker.workeroptions/#activities) object. 
Usually, that's the Activity function's name. 
For example, in the following code, the Activity Type is `greet`:

<!--SNIPSTART typescript-activity-fn -->
<!--SNIPEND-->

<!--SNIPSTART typescript-worker-create -->
<!--SNIPEND-->

You can customize the Activity Type by choosing a property other than the function name. In the following example, the Activity Type is `activityFoo`:

<!--SNIPSTART typescript-custom-activity-type -->
<!--SNIPEND-->
```

File: docs-src/app-dev-context/customize-workflow-type.md [hunk 3]
```diff
@@ -2,9 +2,11 @@
 id: customize-workflow-type
 title: How to customize your Workflow Type
 description: You can set a custom name for your Workflow Type.
-sidebar_label: Customize Workflow Type
+sidebar_label: Workflow Type
```
Comment (flossypurse) [inline docs-src/app-dev-context/customize-workflow-type.md:5]: We are adding "Customize" to the Activity Type one - why remove it here?

i.e -> Customize Activity Type
-> Customize Workflow Type

Comment (flossypurse) [reply docs-src/app-dev-context/customize-workflow-type.md:5]: Ya, I don't mind one way or the other here actually - just that we have the same for each one.

**Review by flossypurse — APPROVED** [review]

**Pull request merged**